docs/src/main/asciidoc/_client-guide/authentication-client.adoc

```diff
@@ -292,6 +292,47 @@ The individual match elements are: -
 * <match-domain-name name="..."/>:: Matches where the protocol of the URI is 'domain' and the scheme specific part of the URI is the name specified within this match element.
  <match-abstract-type name="..." authority="..." />:: Matches the abstract type and/or authority against the values specified within this match element.
 
+=== Elytron client default SSL context security provider
+
+You can register the `org.wildfly.security.auth.client.WildFlyElytronClientDefaultSSLContextProvider` security provider when you have runtime dependency on wildfly-elytron-client. This java security provider loads Elytron client configuration and provides an initialized default SSL context. You can make use of this SSL context by calling `SSLContext.getDefault()` if this provider is registered with high enough priority.
+
+The provider loads the SSL context from either current authentication context obtained from classpath, or from authentication context obtained from file whose path is passed into the security provider either programmatically or as an argument in `java.security` file. Argument passed to the provider directly has precedence over authentication context from classpath.
+
+The SSL context configured to match all rules is the one that will be initialized and returned by this provider:
+
+[source,xml,options="nowrap"]
+----
+<rule use-ssl-context="ssl-context-for-client-provider" />
+----
+
+Or:
+
+----
+myAuthenticationContext.withSsl(MatchRule.ALL, myDefaultSslContext);
+----
+
+Example of dynamic configuration of client default SSL context provider with configuration file path provided:
```

Comment:
Maybe this could be reworded to:

To register the provider programmatically and specify the path to an Elytron client configuration file, the following code can be used:

Reply:
Fixed